static/docs/commands-reference/push.md

```diff
@@ -1,42 +1,115 @@
 # push
 
-This command pushes all data file caches related to the current Git branch to
-the remote storage.
+Uploads files and directories from the current branch in the local workspace to
+the [remote storage]('doc/commands-reference/remote').
 
-The command pushes only output of a specific stage if dvc file is specified `dvc push data.zip.dvc`.
+## Synopsis
 
+```usage
+    usage: dvc push [-h] [-q | -v] [-j JOBS] [--show-checksums] 
+                [-r REMOTE] [-a]
+                [-T] [-d] [-R]
+                [targets [targets ...]]
+
+    positional arguments:
+      targets               DVC files.
+
+```
+
+## Description
+
+The `dvc push` command is the twin pair to the `dvc pull` command, and together
```

Comment:
I see that a lot of users still don't understand how all these commands work (`dvc status -c`, `dvc pull/push/fetch`, etc). Could we think about some explanation similar to what we have in `dvc add`? It might be helpful to try making examples more detailed - show DVC file content, explain that it will extract checksums from it and will be pushing/pulling only those files to/from cache to/from remote.